### PR TITLE
[WebHook]Add option to disable ngrok

### PR DESCRIPTION
**Pull request by valouvaliavlo**

### CODE REVIEW

File: Services/Services_bases/webhook_service/webhook.py [hunk 1]
```diff
@@ -68,14 +79,20 @@ def is_setup_correctly(config):
     def get_is_enabled(config):
         return True
 
+    def check_required_config(self, config):
+        try:
+            return False if config[WebHookService.CONFIG_NGROK] \
+                and configuration.has_invalid_default_config_value(config[services_constants.CONFIG_NGROK_TOKEN]) else True
```
Comment (Herklos) [inline Services/Services_bases/webhook_service/webhook.py:85]: :+1:

File: Services/Services_bases/webhook_service/webhook.py [hunk 2]
```diff
@@ -23,29 +23,40 @@
...
     CONNECTION_TIMEOUT = 3
+    CONFIG_NGROK = "ngrok"
+    CONFIG_WEBHOOK_SERVER_IP = "webhook-bind-ip"
+    CONFIG_WEBHOOK_SERVER_PORT = "webhook-bind-port"
     LOGGERS = ["pyngrok.ngrok", "werkzeug"]
 
     def get_fields_description(self):
         return {
-            services_constants.CONFIG_NGROK_TOKEN: "The ngrok token used to expose the webhook to the internet."
+            WebHookService.CONFIG_NGROK: "Use Ngrok",
```
Comment (Herklos) [inline Services/Services_bases/webhook_service/webhook.py:41]: Can you add these constants in OctoBot-Services ?

File: Services/Services_bases/webhook_service/webhook.py [hunk 3]
```diff
@@ -23,29 +23,40 @@
...
+            services_constants.CONFIG_NGROK_TOKEN: "",
+            WebHookService.CONFIG_WEBHOOK_SERVER_IP: services_constants.DEFAULT_WEBHOOK_SERVER_IP,
+            WebHookService.CONFIG_WEBHOOK_SERVER_PORT: services_constants.DEFAULT_WEBHOOK_SERVER_PORT
         }
 
     def __init__(self):
         super().__init__()
         self.ngrok_tunnel = None
         self.webhook_public_url = ""
+        self.ngrok = True
```
Comment (Herklos) [inline Services/Services_bases/webhook_service/webhook.py:59]:
```suggestion
        self.ngrok_enabled = True
```

File: Services/Services_bases/webhook_service/webhook.py [hunk 4]
```diff
@@ -23,29 +23,40 @@
...
+            WebHookService.CONFIG_NGROK: "Use Ngrok",
+            services_constants.CONFIG_NGROK_TOKEN: "The ngrok token used to expose the webhook to the internet.",
+            WebHookService.CONFIG_WEBHOOK_SERVER_IP: "WebHook bind IP",
+            WebHookService.CONFIG_WEBHOOK_SERVER_PORT: "WebHook port"
         }
 
     def get_default_value(self):
         return {
-            services_constants.CONFIG_NGROK_TOKEN: ""
+            WebHookService.CONFIG_NGROK: True,
```
Comment (Herklos) [inline Services/Services_bases/webhook_service/webhook.py:49]:
```suggestion
            WebHookService.CONFIG_ENABLE_NGROK: True,
```

**Review by Herklos — COMMENTED** [review]
Good idea :+1:

File: Services/Services_bases/webhook_service/webhook.py [hunk 5]
```diff
@@ -155,20 +172,23 @@ def webhook(webhook_name):
...
+                                                                                    [WebHookService.CONFIG_NGROK]
+        if self.ngrok:
+            ngrok.set_auth_token(
+                self.config[services_constants.CONFIG_CATEGORY_SERVICES][services_constants.CONFIG_WEBHOOK][
+                    services_constants.CONFIG_NGROK_TOKEN])
         try:
             self.webhook_host = os.getenv(services_constants.ENV_WEBHOOK_ADDRESS,
                                           self.config[services_constants.CONFIG_CATEGORY_SERVICES]
-                                          [services_constants.CONFIG_WEBHOOK][services_constants.CONFIG_WEB_IP])
+                                          [services_constants.CONFIG_WEBHOOK][WebHookService.CONFIG_WEBHOOK_SERVER_IP])
```
Comment (GuillaumeDSM) [inline Services/Services_bases/webhook_service/webhook.py:184]: 👍

File: Services/Services_bases/webhook_service/webhook.py (hunk 2 above)
Comment (valouvaliavlo) [reply Services/Services_bases/webhook_service/webhook.py:41]: yes, https://github.com/Drakkar-Software/OctoBot-Services/pull/103

**Review by GuillaumeDSM — APPROVED** [review]

**Review by Herklos — APPROVED** [review]
Cool feature :+1:

**Pull request merged**